### PR TITLE
Update Go packages

### PR DESCRIPTION
**Pull request by dhaiducek**
Addresses:
- https://github.com/stolostron/backlog/issues/19717

### CODE REVIEW

**Comment by dhaiducek** [comment]
(This is going to break. Need to work out a kink.)

File: go.mod [hunk 1]
```diff
@@ -4,62 +4,54 @@ go 1.17
 
 require (
 	github.com/onsi/ginkgo v1.16.5
```
Comment (JustinKuli) [inline go.mod:6]: If nothing uses ginkgo v1, should this still be here?

File: go.mod [hunk 2]
```diff
@@ -4,62 +4,54 @@ go 1.17
...
+	open-cluster-management.io/api v0.6.0 // indirect
+	open-cluster-management.io/multicloud-operators-subscription v0.6.0 // indirect
+	sigs.k8s.io/controller-runtime v0.11.1 // indirect
+	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace k8s.io/client-go => k8s.io/client-go v0.21.3
+replace k8s.io/client-go => k8s.io/client-go v0.23.2
```
Comment (JustinKuli) [inline go.mod:57]: Is there a 0.23.3 version of this that would match k8s.io/api ?

**Comment by ycao56** [comment]
Do we still generate junit test report after upgrading to v2? This is being used by cicd team to generate canary report. The custom reporter is used to generate it. I think v2 has changed it.

File: go.mod (hunk 2 above)
Comment (dhaiducek) [reply go.mod:57]: Shoot. You're right. Not sure how I got `v0.23.2` stuck in my head. I'll fix it (and then update all the other repos I mis-replaced 🤦 )

File: go.mod (hunk 1 above)
Comment (dhaiducek) [reply go.mod:6]: No 😞 I'll dig into it

**Comment by dhaiducek** [comment]
@ycao56 The test run passed and delivered XML reports (that should be JUnit). I'm not sure how these reports compare to reports previously, but here they are for reference/verification (I'll try to dig into it tomorrow to compare, too):
- https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/stolostron_governance-policy-framework/262/pull-ci-stolostron-governance-policy-framework-main-test-e2e-grc-framework/1495938248323633152/artifacts/test-e2e-grc-framework/e2e/artifacts/

**Comment by ycao56** [comment]
check with @Kyl-Bempah, he should be able to verify.

**Comment by dhaiducek** [comment]
CICD confirmed and will contact us if updates should be made--this PR can be merged.

**Review by ycao56 — APPROVED** [review]
Great job👍

**Pull request merged**